### DISCUSSION TITLE
Can I get the x,y coordinates (in space) of elements in the colour-checker?

### DISCUSSION THREAD

**Discussion by sapir-mardan** [colour-science/colour-checker-detection#82] [Q&A]
I have a bunch of images with colour-checker. I need to find out the coordinates of 4 points (let's say, the middle of each corner-colour) in order to scale what is the size of the soundings in the image. Any way to do so automatically? Many thanks!!

**Comment by sapir-mardan** [comment]
To give an example, if I have the following image, can I get the coordinates of the corners (roughly black, brown, white, light blue) as such:
Image: 020_C.JPG, Colour: DarkSkin, X: 2248, Y: 3317
Image: 020_C.JPG, Colour: Cyan, X: 1236, Y: 3311
Image: 020_C.JPG, Colour: Black, X: 1251, Y: 2660
Image: 020_C.JPG, Colour: White, X: 2256, Y: 2659

<img width="612" alt="image" src="https://github.com/colour-science/colour-checker-detection/assets/92859243/0dec8f1b-01b3-45b2-9b9b-022aa3e20835">

**Reply by KelSolaar** [reply]
Hello,

The current detection methods both output a `tuple` of `DataDetectionColourChecker` class instance when passing `additional_data=True`. This class has a `quadrilateral` attribute that should give you the 4 corners of the detected colour checker: https://github.com/colour-science/colour-checker-detection/blob/develop/colour_checker_detection/detection/common.py#L988

That should hopefully give you what you need!

Cheers,

Thomas

**Reply by jingweileica** [reply]
Hi Thomas, 
I checked the newest code and the coordinates of the 4 points are not exactly the colour checker points that on the original image, are there any relation between the points that given from quadrilateral attribute and the original groundtruth ? 
e.g.

image size 4096 * 3072
on quadrilateral:
array([[ 920.,  680.],
       [ 928.,  825.],
       [ 718.,  837.],
       [ 709.,  692.]], dtype=float32)
       
 ground truth:
 point start from upper left of cc anticlockwise
 array([[ 2044.,  1980.],
       [ 2056.,  2368.],
       [ 2652.,  2340.],
       [ 2588.,  1972.]], dtype=float32)

thanks
